Sprint_2

Tatianabaitt opened the pull request:
Проект содержит основное и дополнительное задание, отчёт в jococo

Tatianabaitt committed:
Sprint_2
Проект содержит основное и дополнительное задание, отчёт в jococo

pomegranate282922 commented on src/test/java/AnimalTest.java:
```diff
@@ -0,0 +1,31 @@
+        } catch (Exception e) {
+            actual = e.getMessage();
+        }
+        assertEquals(message, expected, actual);
```
Так можно, но можно и используя специальные возможности junit4. Посмотри, например, тут: https://jsparrow.github.io/rules/replace-j-unit-expected-exception.html

pomegranate282922 commented on src/test/java/CatTest.java:
```diff
@@ -0,0 +1,48 @@
+    @Test
+    public void getFoodCatReturnsCorrectValue() throws Exception {
+        Cat cat = new Cat(feline);
+        Mockito.when(predator.eatMeat()).thenReturn(List.of("Животные", "Птицы", "Рыба"));
```
Так, тут закралась ошибочка. Смотри - ты создала поле predator, замокировала его, что-то подменяешь в нем. Но! Оно нигде у тетя не используется в объекте, который ты тестируешь (lion)! Значит, все эти действия никакого отношения к тестируемому методу не имеют!
Поправь, пожалуйста!

pomegranate282922 commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,58 @@
+        } catch (Exception e) {
+            actual = e.getMessage();
+        }
+        assertEquals(message, expected, actual);
```
Здесь то же замечание, да?

pomegranate282922 reviewed (commented):
Работа ну почти зачётная - надо только подправить ошибку.

Tatianabaitt committed:
Sprint_2
Заменила поимку и проверку исключений на assertThrows().
Убрала ненужный мок Predator.

pomegranate282922 reviewed (approved):
Отлично!
Единственное отмечу ещё - коммить только то, что необходимо! Папка src и помник. Остальное - по требованию задания. Потому что другое не только не нужно, но и может содержать приватную информацию.
